### ISSUE TITLE
CompileWidgetsetTask not working on 0.10

### ISSUE THREAD

**Issue by jorgebsa** [johndevs/gradle-vaadin-plugin#196]
Hi John,

I've been using your plugin for some months now, and have to say it is very useful indeed, thanks for your contribution. Today, i tried to build one of the Vaadin projects I work on and an error occurred during the CompileWidgetsetTask.

I have been working with version 0.9.8 of your plugin and gradle wrapper 2.2.1 for the past months. However, the build.gradle i was using, was applying the latest version of your plugin. Therefore, today 0.10 was downloaded. 

This is what i got using ./gradlew clean build --debug

````
15:40:53.061 [ERROR] [org.gradle.BuildExceptionReporter] FAILURE: Build failed with an exception.
15:40:53.063 [ERROR] [org.gradle.BuildExceptionReporter] 
15:40:53.068 [ERROR] [org.gradle.BuildExceptionReporter] * Where:
15:40:53.068 [ERROR] [org.gradle.BuildExceptionReporter] Script 'http://plugins.jasoft.fi/vaadin.plugin' line: 55
15:40:53.069 [ERROR] [org.gradle.BuildExceptionReporter] 
15:40:53.070 [ERROR] [org.gradle.BuildExceptionReporter] * What went wrong:
15:40:53.070 [ERROR] [org.gradle.BuildExceptionReporter] A problem occurred evaluating script.
15:40:53.071 [ERROR] [org.gradle.BuildExceptionReporter] > Failed to apply plugin [class 'fi.jasoft.plugin.GradleVaadinPlugin']
15:40:53.072 [ERROR] [org.gradle.BuildExceptionReporter]    > Could not create task of type 'CompileWidgetsetTask'.
15:40:53.073 [ERROR] [org.gradle.BuildExceptionReporter] 
15:40:53.073 [ERROR] [org.gradle.BuildExceptionReporter] * Try:
15:40:53.074 [ERROR] [org.gradle.BuildExceptionReporter] Run with --stacktrace option to get the stack trace. 
````

and this is what i got using --stacktrace instead of --debug
````
org.gradle.api.GradleScriptException: A problem occurred evaluating script.
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:54)
        at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:148)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyScript(DefaultObjectConfigurationAction.java:93)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$000(DefaultObjectConfigurationAction.java:36)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$1.run(DefaultObjectConfigurationAction.java:62)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.java:131)
        at org.gradle.groovy.scripts.DefaultScript.apply(DefaultScript.java:101)
        at org.gradle.api.Script$apply.callCurrent(Unknown Source)
        at vaadin_artifact_2l827ylwjidoe9550thqvzj5g.run(http://gradle.uppoints.com/vaadin-artifact.gradle:9)
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:52)
        at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:148)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyScript(DefaultObjectConfigurationAction.java:93)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$000(DefaultObjectConfigurationAction.java:36)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$1.run(DefaultObjectConfigurationAction.java:62)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.java:131)
        at org.gradle.api.internal.project.AbstractPluginAware.apply(AbstractPluginAware.java:37)
        at org.gradle.api.Project$apply.call(Unknown Source)
        at org.gradle.api.internal.project.ProjectScript.apply(ProjectScript.groovy:34)
        at org.gradle.api.Script$apply.callCurrent(Unknown Source)
        at build_7buaqxvg2gmc62nam9yuao3zi.run(/Volumes/untitled/workspace-java/upp-pos-management-ui/build.gradle:2)
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:52)
        at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:148)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:39)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:26)
        at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:34)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:59)
        at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:504)
        at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:83)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:42)
        at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:35)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:129)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:80)
        at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
        at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:36)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:51)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:171)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
        at org.gradle.launcher.Main.doAction(Main.java:33)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:127)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:56)
Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin [class 'fi.jasoft.plugin.GradleVaadinPlugin']
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyPlugin(DefaultObjectConfigurationAction.java:103)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$100(DefaultObjectConfigurationAction.java:36)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$2.run(DefaultObjectConfigurationAction.java:71)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.java:131)
        at org.gradle.api.internal.project.AbstractPluginAware.apply(AbstractPluginAware.java:37)
        at org.gradle.api.Project$apply.call(Unknown Source)
        at vaadin_2lvlx61gbeu2j21dso5hn2q7a.run(http://plugins.jasoft.fi/vaadin.plugin:55)
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:52)
        ... 55 more
Caused by: org.gradle.api.tasks.TaskInstantiationException: Could not create task of type 'CompileWidgetsetTask'.
        at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:126)
        at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:121)
        at org.gradle.api.internal.AbstractTask.injectIntoNewInstance(AbstractTask.java:138)
        at org.gradle.api.internal.project.taskfactory.TaskFactory.createTaskObject(TaskFactory.java:121)
        at org.gradle.api.internal.project.taskfactory.TaskFactory.createTask(TaskFactory.java:81)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory.createTask(AnnotationProcessingTaskFactory.java:99)
        at org.gradle.api.internal.project.taskfactory.DependencyAutoWireTaskFactory.createTask(DependencyAutoWireTaskFactory.java:39)
        at org.gradle.api.internal.tasks.DefaultTaskContainer.create(DefaultTaskContainer.java:52)
        at org.gradle.api.tasks.TaskContainer$create$0.call(Unknown Source)
        at fi.jasoft.plugin.GradleVaadinPlugin.apply(GradleVaadinPlugin.groovy:86)
        at fi.jasoft.plugin.GradleVaadinPlugin.apply(GradleVaadinPlugin.groovy)
        at org.gradle.api.internal.plugins.DefaultPluginContainer.providePlugin(DefaultPluginContainer.java:188)
        at org.gradle.api.internal.plugins.DefaultPluginContainer.addPluginInternal(DefaultPluginContainer.java:137)
        at org.gradle.api.internal.plugins.DefaultPluginContainer.apply(DefaultPluginContainer.java:107)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyPlugin(DefaultObjectConfigurationAction.java:101)
        ... 62 more
Caused by: java.lang.NoClassDefFoundError: org/gradle/jvm/tasks/Jar
        at fi.jasoft.plugin.tasks.CompileWidgetsetTask.<init>(CompileWidgetsetTask.groovy:116)
        at fi.jasoft.plugin.tasks.CompileWidgetsetTask_Decorated.<init>(Unknown Source)
        at org.gradle.api.internal.DependencyInjectingInstantiator.newInstance(DependencyInjectingInstantiator.java:48)
        at org.gradle.api.internal.ClassGeneratorBackedInstantiator.newInstance(ClassGeneratorBackedInstantiator.java:36)
        at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:124)
        ... 76 more
Caused by: java.lang.ClassNotFoundException: org.gradle.jvm.tasks.Jar
        ... 81 more
````

Any thoughts on this? Is it something wrong on my side? Such as incompatible gradle version with 0.10 or something else? For the time being, i will revert to 0.9.8, but hopefully i can keep up-to-date with the latest version.

Thanks!

**Comment by marcotuliotm** [comment]
+1

**Comment by christoph-frick** [comment]
possible duplicate of #192

**Comment by johndevs** [comment]
The problem seems to be that **0.10** got compiled with Gradle 2.4 which no longer is backward compatible with Gradle 2.2 (**org.gradle.jvm.tasks.Jar** was added in Gradle 2.3). 

To fix the issue i've now made the plugin compile against Gradle 2.0 (7f36d789337d682) which should solve this issue. The fix should already be in the **0.10-SNAPSHOT**.

I'll release a maintenance version soon with that fix but in the mean time either using the snapshot or upgrading your Gradle distribution should fix the issue.

**Comment by johndevs** [comment]
0.10.1 is now released which should fix the ClassNotFound exception.

Please re-open if the problem still persists.
